frontend/public/components/factory/table.tsx

```diff
@@ -124,16 +136,22 @@ const stateToProps = ({UI}, {data = [], defaultSortField = 'metadata.name', defa
 
   if (loaded) {
     let sortBy: string | Function = 'metadata.name';
-    if (currentSortField) {
+    if (currentSortField && !currentSortFunc) {
       // Sort resources by one of their fields as a string
       sortBy = resource => sorts.string(_.get(resource, currentSortField, ''));
-    } else if (currentSortFunc && sorts[currentSortFunc]) {
+    } else if (!currentSortField && currentSortFunc && sorts[currentSortFunc]) {
       // Sort resources by a function in the 'sorts' object
       sortBy = sorts[currentSortFunc];
     }
 
-    // Always set the secondary sort criteria to ascending by name
-    newData = _.orderBy(newData, [sortBy, 'metadata.name'], [currentSortOrder, SortByDirection.asc]);
+    if (currentSortField && currentSortFunc && sorts[currentSortFunc]) {
+      // if sortFld & sortFunc, pass sortField to sortFunc
```

Comment:
Instead of repurposing `sortFunc` for this, I'd probably make this a different prop that just accepts a function directly. You're changing the behavior of `sortFunc` depending on another prop, which is confusing.

Reply:
Specifically, I'm thinking something like

```jsx
sortField="spec.myproperty" sortConversion={_.toNumber}
```

Then you don't have to create a special chargeback `sortFunc`

Reply:
@spadgett, Ok I'll revert the sorting changes.  Not sure what diff between a `sortConversion` and `sortFunc` would be?  Both would have to be processed/persisted via [table stateToprops](https://github.com/openshift/console/blob/master/frontend/public/components/factory/table.tsx#L122) `orderBy` sorting method.

Agree it's better to wait for the v2 api to know the actual types :+1: